src/i_midimusic.c

```diff
@@ -280,16 +317,30 @@ static void ResetPitchBendSensitivity(void)
     for (i = 0; i < MIDI_CHANNELS_PER_TRACK; ++i)
     {
         // Set RPN MSB/LSB to pitch bend sensitivity.
-        SendShortMsg(MIDI_EVENT_CONTROLLER, i, MIDI_CONTROLLER_RPN_LSB, 0);
-        SendShortMsg(MIDI_EVENT_CONTROLLER, i, MIDI_CONTROLLER_RPN_MSB, 0);
+        SendControlChange(i, MIDI_CONTROLLER_RPN_LSB, MIDI_RPN_PITCH_BEND_SENS_LSB);
+        SendControlChange(i, MIDI_CONTROLLER_RPN_MSB, MIDI_RPN_MSB);
 
         // Reset pitch bend sensitivity to +/- 2 semitones and 0 cents.
-        SendShortMsg(MIDI_EVENT_CONTROLLER, i, MIDI_CONTROLLER_DATA_ENTRY_MSB, 2);
-        SendShortMsg(MIDI_EVENT_CONTROLLER, i, MIDI_CONTROLLER_DATA_ENTRY_LSB, 0);
+        SendControlChange(i, MIDI_CONTROLLER_DATA_ENTRY_MSB, 2);
+        SendControlChange(i, MIDI_CONTROLLER_DATA_ENTRY_LSB, 0);
 
         // Set RPN MSB/LSB to null value after data entry.
-        SendShortMsg(MIDI_EVENT_CONTROLLER, i, MIDI_CONTROLLER_RPN_LSB, 127);
-        SendShortMsg(MIDI_EVENT_CONTROLLER, i, MIDI_CONTROLLER_RPN_MSB, 127);
+        SendControlChange(i, MIDI_CONTROLLER_RPN_LSB, MIDI_RPN_NULL);
+        SendControlChange(i, MIDI_CONTROLLER_RPN_MSB, MIDI_RPN_NULL);
+    }
+}
+
+// Wait time based on bytes sent to the device. For compatibility with hardware
+// devices (e.g. SC-55).
+
+static void ResetDelayBytes(uint32_t length)
+{
+    if (midi_reset_delay == -1)
+    {
+        // MIDI transfer period is 320 us per byte (MIDI 1.0 Electrical Spec
+        // Update CA-033, page 2). Round up value.
+        const uint32_t delay = length * 320 / 1000 + (length * 320 % 1000 != 0);
+        SDL_Delay(delay);
```

Comment:
How about this?
```diff
diff --git a/src/i_midimusic.c b/src/i_midimusic.c
index 40badcc..8c4c8c2 100644
--- a/src/i_midimusic.c
+++ b/src/i_midimusic.c
@@ -376,8 +376,7 @@ static void ResetDelayBytes(uint32_t length)
     {
         // MIDI transfer period is 320 us per byte (MIDI 1.0 Electrical Spec
         // Update CA-033, page 2). Round up value.
-        const uint32_t delay = length * 320 / 1000 + (length * 320 % 1000 != 0);
-        SDL_Delay(delay);
+        I_SleepUS(320 * length);
     }
 }
 
```
I like the auto delay option, it seems to help with the "popping" problem.

Reply:
> How about this?

Oh yeah, I forgot about that function, thanks. Fixed in https://github.com/fabiangreffrath/woof/pull/1657/commits/223ebd0b5e492cf4c8f39e6bd47e65ddb60d15a8.

>I like the auto delay option, it seems to help with the "popping" problem.

It seems to work well so far.